scipy/optimize/_quasi_newton_approx.py

```diff
@@ -0,0 +1,261 @@
+        result and the unmodified matrix. Both methods are explained
+        in [1]_, p.536-537.
+    min_curvature : float
+        Define the minimum curvature ``dot(delta_grad, delta_x)``
+        allowed to go unnafected by the exception strategy. By default
+        is equal to 1e-2 when ``exception_strategy = 'skip_update'``
+        and equal to 0.2 when ``exception_strategy = 'damped_bfgs'``.
+
+    References
+    ----------
+    .. [1] Nocedal, Jorge, and Stephen J. Wright. "Numerical optimization"
+           Second Edition (2006).
+    """
+
+    def __init__(self, exception_strategy='skip_update', min_curvature=None):
+        if exception_strategy == 'skip_update':
+            if min_curvature is not None:
+                self.min_curvature = min_curvature
+            else:
+                self.min_curvature = 1e-2
+        elif exception_strategy == 'damped_bfgs':
+            if min_curvature is not None:
+                self.min_curvature = min_curvature
+            else:
+                self.min_curvature = 0.2
+        else:
+            ValueError("Unknown approx_type.")
+        super(BFGS, self).__init__()
+        self.exception_strategy = exception_strategy
+
+    def _update_inverse_hessian(self, ys, Hy, yHy, s):
+        """Update inverse Hessian matrix.
+
+        BFGS update using the formula:
+
+            ``H <- H + ((H*y).T*y + s.T*y)/(s.T*y)^2 * (s*s.T) - 1/(s.T*y) * ((H*y)*s.T + s*(H*y).T)``
+
+        where ``s = delta_x`` and ``y = delta_grad``. This formula is
+        equivalent to (6.17) in [1]_ written in a more efficient way
+        for implementation.
+
+        References
+        ----------
+        .. [1] Nocedal, Jorge, and Stephen J. Wright. "Numerical optimization"
+               Second Edition (2006).
+        """
+        self.H = self._syr2(-1.0 / ys, s, Hy, a=self.H)
+        self.H = self._syr((ys+yHy)/ys**2, s, a=self.H)
+        return
+
+    def _update_hessian(self, ys, Bs, sBs, y):
+        """Update Hessian matrix.
+
+        BFGS update using the formula:
+
+            ``B <- B - (B*s)*(B*s).T/s.T*(B*s) + y*y^T/s.T*y``
+
+        where ``s`` is short for ``delta_x`` and ``y`` is short
+        for ``delta_grad``. Formula (6.19) in [1]_.
+
+        References
+        ----------
+        .. [1] Nocedal, Jorge, and Stephen J. Wright. "Numerical optimization"
+               Second Edition (2006).
+        """
+        self.B = self._syr(1.0 / ys, y, a=self.B)
+        self.B = self._syr(-1.0 / sBs, Bs, a=self.B)
+        return
+
+    def update(self, delta_x, delta_grad):
+        """Update approximation matrix.
+
+        Parameters
+        ----------
+        delta_x : ndarray
+            The difference between two points the gradient
+            function have been evaluated: ``delta_x = x2 - x1``.
+        delta_grad : ndarray
+            The difference between the gradient evaluated
+            in two points: ``delta_grad = grad(x2) - grad(x1)``.
```

Comment:
I guess "at the two points".

Reply:
I opt to rewrite the phrase:

 `The difference between the gradients: ``delta_grad = grad(x2) - grad(x1)``.

I think it sounds better